utils/add_platform_rule.py

```diff
@@ -0,0 +1,383 @@
+#!/usr/bin/python3
+
+import argparse
+import subprocess
+import pathlib
+import sys
+import textwrap
+import os
+import time
+
+PROG_DESC = (''' Create and test content files for Kubernetes API checks.
+
+This script is intended to help content writers create a new application check
+for OCP4/Kubernetes.
+
+- The 'create' subcommand creates the initial files for a new rule and fetches
+  the raw URL of the object in question (unless you specify the URL).
+
+- The 'test' subcommand builds your content locally and tests directly using an
+  openscap podman container. The scan container will test against yaml files
+  staged under --objectdir.
+
+- The 'cluster-test' subcommand pushes the content to your cluster, and then
+  runs a Platform scan for your rule with compliance-operator.
+
+Example workflow:
+
+$ utils/add_platform_rule.py create --rule=ocp_proxy_has_ca \
+  --type="proxies.config" --name="cluster" \
+  --yamlpath=".spec.trustedCA.name" --match="[a-zA-Z0-9]*"
+creating check for "/apis/config.openshift.io/v1/proxies/cluster" with yamlpath ".spec.trustedCA.name" satisfying match of "[a-zA-Z0-9]*"
+wrote applications/openshift/ocp_proxy_has_ca/rule.yml
+wrote applications/openshift/ocp_proxy_has_ca/oval/shared.xml
+$ mkdir -p /tmp/apis/config.openshift.io/v1/proxies/
+$ oc get proxies.config/cluster -o yaml > /tmp/apis/config.openshift.io/v1/proxies/cluster
+$ utils/add_platform_rule.py test --rule=ocp_proxy_has_ca
+testing rule ocp_proxy_has_ca locally
+Title
+        None
+Rule
+        xccdf_org.ssgproject.content_rule_ocp_proxy_has_ca
+Ident
+        CCE-84209-6
+Result
+        pass
+
+$ utils/add_platform_rule.py cluster-test --rule=ocp_proxy_has_ca
+testing rule ocp_proxy_has_ca in-cluster
+deploying compliance-operator
+pushing image build to cluster
+waiting for cleanup from previous test run
+output from last phase check: LAUNCHING NOT-AVAILABLE
+output from last phase check: RUNNING NOT-AVAILABLE
+output from last phase check: AGGREGATING NOT-AVAILABLE
+output from last phase check: DONE COMPLIANT
+COMPLIANT
+
+''')
+
+PLATFORM_RULE_DIR = 'applications/openshift'
+OSCAP_TEST_IMAGE = 'quay.io/compliance-operator/openscap-ocp:1.3.3'
+OSCAP_CMD_OPTS = 'oscap xccdf eval --fetch-remote-resources --profile xccdf_org.ssgproject.content_profile_test --results-arf /tmp/report-arf.xml /content/ssg-ocp4-ds.xml'
+PROFILE_PATH = 'ocp4/profiles/test.profile'
+
+RULE_TEMPLATE = ('''prodtype: ocp4
+
+title: {TITLE}
+
+description: TBD
+
+rationale: TBD
+
+identifiers:
+    {IDENT}
```

Comment:
This wouldn't be a valid type for identifiers. a dictionary of identifiers is what's really expected here. IMO you should just leave this as an empty dict in yaml, and let the user edit it later. If not, somehow you'd need to parse the dict of identifiers.